spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java

```diff
@@ -508,13 +522,39 @@ public void testUnindexedField() {
 
 	@Test
 	public void testEmbeddedEntity() {
-		List<TestItemWithEmbeddedEntity.EmbeddedEtity> embeddedEtities = Arrays.asList(
-				new TestItemWithEmbeddedEntity.EmbeddedEtity("item 0"),
-				new TestItemWithEmbeddedEntity.EmbeddedEtity("item 1"));
+		EmbeddedEntity embeddedEntityA = new EmbeddedEntity("item 0");
```

Comment:
do we test that the added exceptions appear when they expected and that they have the right messages?

Reply:
I have read that it is not a good idea to encode the text of exception messages into tests because of the burden of maintaining code like that. 

There aren't any new exceptions being thrown because this is using nearly all the existing mapping/converting code. The exception message about unsupported embedded type is not testable, and is there to catch the case where we add a new type in that enum but forget to account for it elsewhere.

Reply:
You can use the JUnit 4 rule to easily test expected exception message: https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html
Since we don't have a rich exception hierarchy, I think verifying the messages is useful.
I would be interested in reading a description of a best practice to the contrary.